api/server/http2_router.go

```diff
@@ -25,15 +25,14 @@ func newHTTP2Router() *http2Router {
 }
 
 func (h *http2Router) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	// The :path pseudo-header takes the form of /Prefix/Path
-	parsed := strings.Split(r.URL.Path, "/")
-	if len(parsed) < 2 {
+	// the chain-id header must be set to route the request to the correct chain
+	// http2 handler
+	chainID := r.Header.Get("chain-id")
+	if len(chainID) == 0 {
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
-	chainID := parsed[1]
-
```

Comment:
Lower in this file, we still `r.Clone` the context and strip the `/chainID` prefix. This code isn't needed anymore right?